registrations/api.py

```diff
@@ -277,36 +282,28 @@ def filter_queryset(self, queryset):
     def filter_registration(self, queryset, name, value: list[str]):
         user = self.request.user
 
-        # Get admin organizations and descendants only once instead of using is_admin method
-        registration_admin_organizations = (
-            user.get_registration_admin_organizations_and_descendants()
-        )
+        try:
+            registrations = Registration.objects.filter(pk__in=value)
+        except ValueError:
+            registrations = Registration.objects.none()
 
-        registrations = []
-        for pk in value:
-            try:
-                registration = Registration.objects.select_related(
-                    "event__publisher"
-                ).get(pk=pk)
-            except (ValueError, Registration.DoesNotExist):
-                raise ParseError(
-                    _("Registration with id {pk} doesn't exist.").format(pk=pk)
-                )
+        if not registrations:
+            raise ParseError(_("No registrations found with the given IDs."))
 
-            if not (
-                user.is_superuser
-                or user.is_registration_user_access_user_of(
-                    registration.registration_user_accesses
-                )
-                or registration.publisher in registration_admin_organizations
-            ):
-                raise DRFPermissionDenied(
-                    _(
-                        "Only the admins of the organization {organization} have access rights."
-                    ).format(organization=registration.publisher)
-                )
+        if user.is_superuser:
+            return queryset.filter(**{f"{name}__in": registrations})
```

Comment:
I don't think the dynamic nature of `name` is needed, at least in the current state, here or below on line 308. 

```python
return queryset.filter(registration__in=registrations})
```

would be quite a bit more readable (and searchable!).

Reply:
Changed as suggested